### PR TITLE
Async session tracker initialization

### PR DESCRIPTION
**Pull request by andreiko**
This is a follow-up to https://github.com/gravitational/teleport/pull/18302
 
Additional testing showed that connecting to nodes during periods when auth service is down can hang for extra 20-40 seconds which is a bad experience.

The hanging is caused by ssh service trying to create a session tracker using Auth API. It gets better when the breaker trips and all Auth calls start failing immediately, but the delay becomes significant when the breaker starts letting requests through again.

### CODE REVIEW

File: lib/srv/sess.go [hunk 1]
```diff
@@ -1773,8 +1773,12 @@ func (s *session) trackSession(ctx context.Context, teleportUser string, policyS
 		svc = nil
 	}
 
+	// apply shorter timeout to make opening ssh sessions more responsive during periods when auth service is down
+	timeoutCtx, timeoutCancel := context.WithTimeout(ctx, time.Second*5)
```
Comment (andreiko) [inline lib/srv/sess.go:1777]: Couldn't find a suitable value in the `defaults` package or any configuration value accessible from here, hence the hardcoded timeout.

**Review by zmb3 — APPROVED** [review]

**Review by rosstimothy — CHANGES_REQUESTED** [review]
IIRC we agreed in the scale meeting that to fix this we should move the creation of the session tracker into a goroutine if and only if the session is not moderated instead of lowering any timeouts. @fspmarshall @espadolini is that correct?

**Comment by andreiko** [comment]
Yeah, we were discussing multiple options, it's possible that I remembered wrong which one we chose.

I implemented async session tracker initialization such that, if the session is not moderated, `trackSession()` returns immediately and tracker is supposed to be accessed via `getTracker()` which blocks until initialization is complete.

This works perfectly for command execution, but interactive sessions won't start until some tracker is available. In my current version, when auth is down, it blocks [addParty() here](https://github.com/gravitational/teleport/blob/00a0f660c63675938c0f9fbf9786e2f2583db45e/lib/srv/sess.go#L1551) for 30 seconds which often leads to the "timed out waiting for continue signal" error and failure.

Maybe I'm missing something? When you suggested to create session tracker in a goroutine, did you have some different synchronization mechanism in mind?

**Comment by andreiko** [comment]
> If we were to fix #17027 would that resolve the issues you are seeing when blocking in `addParty`?

Yeah, it looks like it would!

**Comment by andreiko** [comment]
Closing for now.

**Pull request closed without merging**